src/Microsoft.Owin/Extensions/AppBuilderCookieExtensions.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Owin.Infrastructure;
+using Owin;
+
+namespace Microsoft.Owin
+{
+    /// <summary>
+    /// Cookie-related extensions for <see cref="IAppBuilder"/>.
+    /// </summary>
+    public static class AppBuilderCookieExtensions
+    {
+        /// <summary>
+        /// Sets the default <see cref="ICookieManager"/> that will be used by
+        /// the OWIN middleware when no manager is explicitly set in their options.
+        /// </summary>
+        /// <remarks>
+        /// Depending on the host, a default <see cref="ICookieManager"/> instance
+        /// may already be registered in <see cref="IAppBuilder.Properties"/>
+        /// (e.g SystemWeb automatically registers a cookie manager that relies on
+        /// HttpCookieCollection instead of directly manipulating the HTTP headers).
+        /// </remarks>
+        /// <param name="app">The application builder.</param>
+        /// <param name="manager">The cookie manager.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static void SetDefaultCookieManager(this IAppBuilder app, ICookieManager manager)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (manager is null)
+            {
+                app.Properties.Remove("infrastructure.CookieManager");
```

Comment:
Use constants for the key names.

Reply:
Done. I also updated the title to better reflect that these extensions can also be used by applications (and not just hosts) to override the default cookie managers.